core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -775,6 +775,14 @@ class DynamicListenerConfig(server: KafkaServer) extends BrokerReconfigurable wi
     }
     if (!newAdvertisedListeners.contains(newConfig.interBrokerListenerName))
       throw new ConfigException(s"Advertised listener must be specified for inter-broker listener ${newConfig.interBrokerListenerName}")
+
+    val endPoints = newConfig.advertisedListeners.map(e => s"${e.host}:${e.port}")
+    server.zkClient.getAllBrokersInCluster.filter(_.id != newConfig.brokerId).foreach(broker => {
```

Comment:
Not sure it's worthwhile given how rarely the listeners would be updated in practice, but I guess we could also use the metadata cache? 

Otherwise, it would be nice to factor out the common logic between this and `KafkaServer`. Like maybe a method in `AdminUtils`?

Reply:
@hachikuji  I am also not strongly inclined to add this check in DynamicConfig validation phase.   DynamicListenerConfig.reconfigure method fails if there any misconfigured listeners. here:
https://github.com/omkreddy/kafka/blob/28bf632dc791ec42077b8ab19aad7583a44975d4/core/src/main/scala/kafka/server/DynamicBrokerConfig.scala#L804

 If there are no concern, I am going to drop this check from DynamicBrokerConfig. validateReconfiguration